streams/src/main/java/org/apache/kafka/streams/StreamsConfig.java

```diff
@@ -641,6 +678,19 @@ public Serde valueSerde() {
         return parsed;
     }
 
+    @Override
+    public Long getLong(final String key) {
+        // return different default value for `commit.interval.ms` if EOS enabled
+        if (StreamsConfig.COMMIT_INTERVAL_MS_CONFIG.equals(key)
+            && eosEnabled
+            // use EOS default value only if no user config is set
+            && !originals().containsKey(StreamsConfig.COMMIT_INTERVAL_MS_CONFIG)) {
+            return EOS_DEFAULT_COMMIT_INTERVAL_MS;
```

Comment:
> I personally prefer to update the config, but I would be nice to do that by hooking into the mechanism that chooses a default instead of mutating the underlying map afterwards.

This sounds like a major rework on `ConfigDef` allowing for "conditional defaults" -- do we really want to have this?